pkg/cluster/clusterserviceprincipal.go

```diff
@@ -173,15 +175,36 @@ func (m *manager) updateAROSecret(ctx context.Context) error {
 }
 
 func (m *manager) updateOpenShiftSecret(ctx context.Context) error {
+	var recreate bool
 	var changed bool
 	spp := m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		//data:
 		// azure_client_id: secret_id
 		// azure_client_secret: secret_value
 		// azure_tenant_id: tenant_id
-		secret, err := m.kubernetescli.CoreV1().Secrets("kube-system").Get(ctx, "azure-credentials", metav1.GetOptions{})
-		if err != nil {
+		secret, err := m.kubernetescli.CoreV1().Secrets(clusterauthorizer.AzureCredentialSecretNameSpace).Get(ctx, clusterauthorizer.AzureCredentialSecretName, metav1.GetOptions{})
+		if kerrors.IsNotFound(err) {
+			// rebuild secret
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterauthorizer.AzureCredentialSecretName,
+					Namespace: clusterauthorizer.AzureCredentialSecretNameSpace,
+				},
+				Type: corev1.SecretTypeOpaque,
+				Data: make(map[string][]byte),
```

Comment:
Why not use the composite literal?
```suggestion
				Data: map[string][]byte{
				     "azure_subscription_id": []byte(m.subscriptionDoc.ID),
				     etc...
				},
```

Reply:
Why not indeed.